schedule/__init__.py

```diff
@@ -214,40 +205,51 @@ class Job(object):
     method, which also defines its `interval`.
     """
 
-    def __init__(self, interval, scheduler=None):
-        self.interval = interval  # pause interval * unit between runs
-        self.latest = None  # upper limit to the interval
-        self.job_func = None  # the job job_func to run
-        self.unit = None  # time units, e.g. 'minutes', 'hours', ...
-        self.at_time = None  # optional time at which this job runs
-        self.last_run = None  # datetime of the last run
-        self.next_run = None  # datetime of the next run
-        self.period = None  # timedelta between runs, only valid for
-        self.start_day = None  # Specific day of the week to start on
-        self.tags = set()  # unique set of tags for the job
-        self.scheduler = scheduler  # scheduler to register with
-
-    def __lt__(self, other):
+    def __init__(self, interval, scheduler: Scheduler):
```

Comment:
Removing the default argument changes the behaviour! Before, you could call this function without a scheduler argument (I'm not sure if that was reasonable, though)

Reply:
You are right, changing this is not good. Will roll this back right away.